File: backend/app/schemas/progress.py
from pydantic import BaseModel


class TopicStat(BaseModel):
    subject: str
    subtopic: str
    total: int
    correct: int
    accuracy: float


class WeakArea(BaseModel):
    subject: str
    subtopic: str
    accuracy: float


class ProgressStats(BaseModel):
    total_questions: int
    total_correct: int
    overall_accuracy: float
    topic_stats: list[TopicStat]
    weak_areas: list[WeakArea]

File: frontend/src/api/progress.ts
const BASE = "/api";

export interface TopicStat {
  subject: string;
  subtopic: string;
  total: number;
  correct: number;
  accuracy: number;
}

export interface ProgressData {
  total_questions: number;
  total_correct: number;
  overall_accuracy: number;
  topic_stats: TopicStat[];
  weak_areas: { subject: string; subtopic: string; accuracy: number }[];
}

export async function getProgress(): Promise<ProgressData> {
  const res = await fetch(`${BASE}/progress`);
  if (!res.ok) throw new Error("Failed to fetch progress");
  return res.json();
}
